### PR TITLE
fix: validate repost item valuation against accounts freeze date

### PR DESCRIPTION
**Pull request by rtdany10**
Peek 2022-11-17 15-33

I have taken into consideration the role allowed to override. Maybe we shouldn't do that?
@deepeshgarg007

#no-docs

### CODE REVIEW

File: erpnext/stock/doctype/repost_item_valuation/repost_item_valuation.py [hunk 1]
```diff
@@ -25,6 +25,25 @@ def validate(self):
...
+	def validate_accounts_freeze(self):
+		acc_settings = frappe.db.get_value(
+			"Accounts Settings",
+			"Accounts Settings",
+			["acc_frozen_upto", "frozen_accounts_modifier"],
+			as_dict=1,
+		)
+		if not acc_settings.acc_frozen_upto:
+			return
+		if acc_settings.frozen_accounts_modifier and self.owner in get_users_with_role(
```
Comment (deepeshgarg007) [inline erpnext/stock/doctype/repost_item_valuation/repost_item_valuation.py]: The doc owner can be forced set to some other owner as well I guess, doesn't seem reliable. Can we check this with the current session user as we do at other places?

**Comment by deepeshgarg007** [comment]
Also, this seems like a `fix` and not a `feat`

**Comment by deepeshgarg007** [comment]
@rtdany10 Any update on this PR?

**Comment by rtdany10** [comment]
> @rtdany10 Any update on this PR?

Sorry for the delay, have changed it to session user.

Also, this:
> I have taken into consideration the role allowed to override. Maybe we shouldn't do that?

File: erpnext/stock/doctype/repost_item_valuation/repost_item_valuation.py (hunk 1 above)
Comment (rtdany10) [reply erpnext/stock/doctype/repost_item_valuation/repost_item_valuation.py]: Updated

**Pull request merged**